playbooks/robusta_playbooks/job_restart_on_oomkilled.py

```diff
@@ -0,0 +1,247 @@
+                        keep_the_same,
+                    )
+                )
+        elif container.name in running_containers:
+            keep_the_same = True
+            containers.append(
+                increase_resource(
+                    container,
+                    max_resource,
+                    params.increase_by,
+                    keep_the_same,
+                )
+            )
+
+    job_spec = job_fields(job_event, containers)
+    job_event.delete()
+    job_spec.create()
+
+    containers_memory_list = []
+
+    # Getting information for finding
+    for index, containers in enumerate(job_spec.spec.template.spec.containers):
+        containers_memory_list.append(containers.name)
+        containers_memory_list.append(containers.resources.requests["memory"])
+
+    finding.add_enrichment(
+        [
+            MarkdownBlock(
+                f"*containers memory after restart*\n```\n{containers_memory_list}\n```"
+            ),
+        ]
+    )
+    event.add_finding(finding)
+
+
+# Function to increase resource of the container
+def increase_resource(container, max_resource, increase_by, keep_the_same):
+    # Getting Container attributes
+    container = Container(
+        name=container.name,
+        image=container.image,
+        livenessProbe=container.livenessProbe,
+        securityContext=container.securityContext,
+        args=container.args,
+        command=container.command,
+        ports=container.ports,
+        lifecycle=container.lifecycle,
+        readinessProbe=container.readinessProbe,
+        workingDir=container.workingDir,
+        env=container.env,
+        startupProbe=container.startupProbe,
+        envFrom=container.envFrom,
+        imagePullPolicy=container.imagePullPolicy,
+        resources=memory_increment(
+            container.resources,
+            increase_by,
+            max_resource,
+            keep_the_same,
+        )
+        if (container.resources.limits and container.resources.requests)
+        else None,
+    )
+    return container
+
+
+# Function to increment in memory
+def memory_increment(resources, increase_by, max_resource, keep_the_same):
+    if keep_the_same:
+        return resources
+    else:
+        requests = bitmath.parse_string_unsafe(resources.requests["memory"])
+        limits = bitmath.parse_string_unsafe(resources.limits["memory"])
+        increase_by = bitmath.parse_string_unsafe(increase_by)
+
+        requests = min((increase_by + requests), max_resource)
+        limits = max(requests, limits)
+
+        formatted_request = requests.format("{value:.0f}{unit}").rstrip("B")
+        formatted_limit = limits.format("{value:.0f}{unit}").rstrip("B")
+
+        return ResourceRequirements(
+            limits={"memory": (str(formatted_limit))},
+            requests={"memory": (str(formatted_request))},
+        )
+
+
+# Function to split number and string from memory[string]
+def split_num_and_str(num_str: str):
```

Comment:
this function is unused and can be removed

Reply:
removed.